pkg/steps/template_test.go

```diff
@@ -238,3 +242,90 @@ func TestOperateOnTemplatePods(t *testing.T) {
 		})
 	}
 }
+
+func TestInjectLabelsToTemplate(t *testing.T) {
+	testCases := []struct {
+		testID   string
+		jobSpec  *api.JobSpec
+		template *templateapi.Template
+		expected *templateapi.Template
+	}{
+		{
+			testID: "nil refs in jobspec, no injection expected",
+			jobSpec: &api.JobSpec{
+				JobSpec: downwardapi.JobSpec{
+					Refs: nil,
+				},
+			},
+			template: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+			},
+			expected: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+			},
+		},
+
+		{
+			testID: "jobspec with refs, label injection expected",
+			jobSpec: &api.JobSpec{
+				JobSpec: downwardapi.JobSpec{
+					Refs: &prowapi.Refs{
+						Org:     "test-org",
+						Repo:    "test-repo",
+						BaseRef: "test-branch",
+					},
+				},
+			},
+			template: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+				Objects: []runtime.RawExtension{
+					func() runtime.RawExtension {
+						pod := &coreapi.Pod{
+							TypeMeta:   meta.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+							ObjectMeta: meta.ObjectMeta{Name: "test-pod"},
+						}
+						return runtime.RawExtension{
+							Raw:    []byte(runtime.EncodeOrDie(corev1Codec, pod)),
+							Object: pod.DeepCopyObject(),
+						}
+					}(),
+				},
+			},
+			expected: &templateapi.Template{
+				TypeMeta:   meta.TypeMeta{Kind: "Template", APIVersion: "template.openshift.io/v1"},
+				ObjectMeta: meta.ObjectMeta{Name: "test-template"},
+				Objects: []runtime.RawExtension{
```

Comment:
I meant _any_ Object field . We just need to check that the template has ObjectLabels set, so all this is extra in every test case

Reply:
oops, yea. I removed the others as well.